### ISSUE TITLE
Build on windows fails

### ISSUE THREAD

**Issue by auto-matic** [ngrok/ngrok-rust#99]
Building the package on windows fails with the following error:
```
   Compiling axum v0.6.18
   Compiling reqwest v0.11.18
   Compiling teloxide-core v0.9.1
   Compiling ngrok v0.12.3
error[E0433]: failed to resolve: could not find `Win32` in `windows_sys`
  --> C:\Users\mrtzk\.cargo\registry\src\index.crates.io-6f17d22bba15001f\ngrok-0.12.3\src\tunnel_ext.rs:24:18
   |
24 | use windows_sys::Win32::Foundation::ERROR_PIPE_BUSY;
   |                  ^^^^^ could not find `Win32` in `windows_sys`

For more information about this error, try `rustc --explain E0433`.
error: could not compile `ngrok` (lib) due to previous error
warning: build failed, waiting for other jobs to finish...
```

**Comment by bobzilladev** [comment]
You can likely pin `mio` in your project's `Cargo.toml` as one way to get around this issue. We'll work on baking that in to the library.
```
# pin mio until all dependencies are also on windows-sys 0.48
# https://github.com/microsoft/windows-rs/issues/2410#issuecomment-1490802715
mio = { version = "=0.8.6" }
```

**Comment by CBenoit** [comment]
`ngrok-rust` requires the feature `Win32_Foundation` from the crate `windows-sys`, but it doesn’t appear to require it explicitly [in the Cargo.toml](https://github.com/ngrok/ngrok-rust/blob/8639d39ef95f03eab97d2cc2b65520a350997d8e/ngrok/Cargo.toml#LL34C1-L34C23).

You can enable it manually until this is addressed:
```toml
windows-sys = { version = "0.45.0", features = ["Win32_Foundation"] }
```

That’s how I fixed the compilation on my side.

**Comment by jrobsonchase** [comment]
:+1: to enabling the feature explicitly. From my reading of the thread, it sounds like we depend on it, but it was only getting incidentally activated by one of our dependencies?

This is one thing I really hate about how features work :upside_down_face: It's not the first time, and probably won't be the last that we've missed one that a dependency, example, test, etc. enabled for us and provided that false sense of security at the standalone crate level.

**Comment by CBenoit** [comment]
> This is one thing I really hate about how features work 🙃

Agreed, that and how hard it is to not duplicate dependencies (here a duplication is caused simply by bumping a minor version of a transitive dependency…). Stuff like [`cargo-deny`](https://embarkstudios.github.io/cargo-deny/checks/bans/diags.html#duplicate) helps with that, but it’s still hard to deal with properly as of now 😕
Anyway, it’s definitely better for `ngrok-rust` itself to enable the feature explicitly and leave the "duplicate hunting" work to downstream user (nothing can be done for that as a library except keeping dependencies up to date…).
